airflow/providers/google/suite/hooks/calendar.py

```diff
@@ -0,0 +1,239 @@
+                service.events()
+                .list(
+                    calendarId=calendar_id,
+                    iCalUID=i_cal_uid,
+                    maxAttendees=max_attendees,
+                    maxResults=max_results,
+                    orderBy=order_by,
+                    pageToken=page_token,
+                    privateExtendedProperty=private_extended_property,
+                    q=q,
+                    sharedExtendedProperty=shared_extended_property,
+                    showDeleted=show_deleted,
+                    showHiddenInvitations=show_hidden_invitation,
+                    singleEvents=single_events,
+                    syncToken=sync_token,
+                    timeMax=time_max,
+                    timeMin=time_min,
+                    timeZone=time_zone,
+                    updatedMin=updated_min,
+                )
+                .execute(num_retries=self.num_retries)
+            )
+            events.extend(response["items"])
+            page_token = response.get("nextPageToken")
+            if not page_token:
+                break
+        return events
+
+    def create_event(
+        self,
+        event: Dict[str, Any],
+        calendar_id: str = 'primary',
+        conference_data_version: Optional[int] = 0,
+        max_attendees: Optional[int] = None,
+        send_notifications: Optional[bool] = False,
+        send_updates: Optional[str] = 'false',
+        supports_attachments: Optional[bool] = False,
+    ) -> dict:
+        """
+        Create event on the specified calendar
+        https://developers.google.com/calendar/api/v3/reference/events/insert
+
+        :param calendar_id: The Google Calendar ID to interact with
+        :type calendar_id: str
+        :param conference_data_version: Optional. Version number of conference data
+            supported by the API client.
+        :type conference_data_version: int
+        :param max_attendees: Optional. If there are more than the specified number of attendees,
+            only the participant is returned.
+        :type max_attendees: int
+        :param send_notifications: Optional. Default is False
+        :type send_notifications: bool
+        :param send_updates: Optional. Default is "false". Acceptable values as "all", "none",
+            "externalOnly"
+        :type send_updates: str
+        :type supports_attachments: Optional. Default is False
+        :type supports_attachments: bool
+        :type event: Required. Request body of Events resource. Start and End are required
+            https://developers.google.com/calendar/api/v3/reference/events#resource
+        :type event: dict
+        :rtype: Dict
+        """
+        if "start" not in event or "end" not in event:
+            raise AirflowException(
+                "start and end must be specified in the event body while creating an event. "
+                "Refer https://developers.google.com/calendar/api/v3/reference/events/insert "
```

Comment:
We probably want to replace `v3` with the api version the user set

Reply:
Updated